M2/Macaulay2/m2/Core.m2

```diff
@@ -120,9 +120,12 @@ needs = filename -> if not filesLoaded#?filename then load filename else (
 -- Setup persistent history
 -----------------------------------------------------------------------------
 historyFilename = "history.m2"
+historyOffset = 0;
 
 if not noinitfile and not gotarg "--no-readline" then (
-    addStartFunction(() -> readHistory(applicationDirectory() | historyFilename));
+    addStartFunction(() -> (
+	    readHistory(applicationDirectory() | historyFilename);
+	    historyOffset = historyLength()));
```

Comment:
I'm actually blanking on why `addStartFunction` was needed here. Why not just do this when `Core` is loading?

Reply:
Yeah, it works just fine without `addStartFunction`.  I just pushed a new commit removing it.